src/gpio.rs

```diff
@@ -1429,6 +1429,16 @@ pub fn enable_isr_service() -> Result<(), EspError> {
     Ok(())
 }
 
+/// Notifies this GPIO driver that the GPIO ISR service has already been initialized.
+/// This prevents an error as this driver would otherwise attempt to initialize it again.
+///
+/// # Safety
+///
+/// The caller must ensure that `gpio_install_isr_service` has already been invoked elsewhere.
+pub unsafe fn isr_service_already_init() {
```

Comment:
I think we should try to document the purpose here a bit better. E.g that this as you say can be used as a "escape hatch". For example if external code calles gpio_install_isr_service or gpio_isr_register. And that should only be needed in that specific cases and never in another situation

Also a naming nit, we could potentially naming it set_isr_service_flag_unchecked. That way we go with the normal rust naming scheme. This is just a idea though.

Reply:
LGTM!